### ISSUE TITLE
Capi issues with webpack

### ISSUE THREAD

**Issue by SaltyCucumber** [paritytech/capi#990]
I'm trying to run capi on a react + TS + webpack stack. Node 18.14.0
and have a bunch of issues:

ERROR in ./node_modules/@capi/westend/chain.js 1814:13-72
export 'PolkadotRuntimeParachainsDisputesPalletCall' (imported as 't') was not found in './types.js'

and a lot of errors similar to this
ERROR in ./node_modules/capi/esm/crypto/hashers.js 71:19-26
export 'Blake2b' (imported as 'Blake2b') was not found in '../deps/wat_the_crypto.js' (module has no exports)

ERROR in node:net
Module build failed: UnhandledSchemeError: Reading from "node:net" is not handled by plugins (Unhandled scheme).
probably missing some polyfill for this one

here is the branch with the code I'm running
https://github.com/SaltyCucumber/assets-portal/pull/86

here you can see errors in-depth:
https://app.netlify.com/sites/assets-portal/deploys/645d21715d53ec000855e7c4

I've used all sorts of hacks and polyfills, so once I'm able to actually see the results I'll trim down all the unnecessary stuff

**Comment by SaltyCucumber** [comment]
have some further info on the bugs and issues

=======================================================================
1. so all of the  '../deps/wat_the_crypto.js' bugs can be fixed if package.json has a minor change:

how it is right now:
```
      "import": {
        "default": "./esm/mod.js",
        "types": "./types/mod.d.ts"
      },
      "require": {
        "default": "./script/mod.js",
        "types": "./types/mod.d.ts"
      }
```

how it should be:

```
      "import": {
        "types": "./types/mod.d.ts",
        "default": "./esm/mod.js"
      },
      "require": {
        "types": "./types/mod.d.ts",
        "default": "./script/mod.js"
      }
```
=======================================================================
2. the issue with PolkadotRuntimeParachainsDisputesPalletCall type in @capi/westend/types.js:
how it is
`export const $polkadotRuntimeParachainsDisputesPalletCall = _codecs.$314`

how similar working export looks like:
```
export const $polkadotRuntimeParachainsDisputesSlashingCall = _codecs.$318

export const PolkadotRuntimeParachainsDisputesSlashingCall = {
  reportDisputeLostUnsigned(value) {
    return C.Rune.object({
      type: "reportDisputeLostUnsigned",
      ...C.RunicArgs.resolve(value),
    })
  },
  isreportDisputeLostUnsigned(value) {
    return value.type === "reportDisputeLostUnsigned"
  },
}
```
how it can be "hacked" right now by adding:
`export const PolkadotRuntimeParachainsDisputesPalletCall = _codecs.$314`

=======================================================================
3. there is another issue with star exports
```
The requested module './pallets.js' contains conflicting star exports for the names 'WestendAssignedSlotsRune', 'WestendAuctionsRune', 'WestendAuthorityDiscoveryRune', 'WestendAuthorshipRune', 'WestendBabeRune', 'WestendBalancesRune', 'WestendConfigurationRune', 'WestendCrowdloanRune', 'WestendDmpRune', 'WestendElectionProviderMultiPhaseRune', 'WestendFastUnstakeRune', 'WestendGrandpaRune', 'WestendHistoricalRune', 'WestendHrmpRune', 'WestendIdentityRune', 'WestendImOnlineRune', 'WestendIndicesRune', 'WestendInitializerRune', 'WestendMultisigRune', 'WestendNominationPoolsRune', 'WestendOffencesRune', 'WestendParaInclusionRune', 'WestendParaInherentRune', 'WestendParaSchedulerRune', 'WestendParaSessionInfoRune', 'WestendParachainsOriginRune', 'WestendParasDisputesRune', 'WestendParasRune', 'WestendParasSharedRune', 'WestendParasSlashingRune', 'WestendParasSudoWrapperRune', 'WestendPreimageRune', 'WestendProxyRune', 'WestendRecoveryRune', 'WestendRegistrarRune', 'WestendSchedulerRune', 'WestendSessionRune', 'WestendSlotsRune', 'WestendStakingRune', 'WestendSudoRune', 'WestendSystemRune', 'WestendTimestampRune', 'WestendTransactionPaymentRune', 'WestendUmpRune', 'WestendUtilityRune', 'WestendVestingRune', 'WestendVoterListRune', 'WestendXcmPalletRune' with the previous requested module './chain.js'
```

dirty hack is to just comment out `export * from "./pallets.js"`, which is of course nowhere near a solution, it's just to go a step further.

=======================================================================
4. Reading from "node:net" is not handled by plugins
regarding this, I didn't find any plugin/polyfill/hack, so not sure how to overcome this one
as Node net library is not browser compatible
will have to try Vite and see how that manages to not have any of the above issues, while webpack has so many problems

**Comment by kratico** [comment]
Issue 1 will be fixed once `wat_the_crypto` is refactored
Issue 2 needs bug fix #1005 
Issue 3 will be fixed in release `v0.1.0-beta.40` with https://github.com/paritytech/capi/commit/353b653a2752e079328e4ee07b38301965d46cbe
Issue 4 may be fixed configuring `node:net` in [externals](https://webpack.js.org/configuration/externals/)
```
// in webpack.config.js
export default {
  externals: {
    "node:net": {},
  },
}
```

@harrysolovay `node:net` is used because `esm/mod.js` imports it following these imports.
```
mod.js > nets/mod.js > nets/DevNetSpec.js > util/port.node.js
```

**Comment by SaltyCucumber** [comment]
> externals: {
    "node:net": {},
  },
  
  oh g, I was so close to this, wrote `net: {}` instead, thought it would work the same principle as with 
`fallback: { crypto: 'crypto-browserify' }`, just with net lib
Thank you very much for the soultion @kratico
